### ISSUE TITLE
Add ability to render view from string

### ISSUE THREAD

**Issue by RomeroMsk** [yiisoft/yii2#7309]
I can't find the way to render a view from string (not from file). I want to use template engine (Twig or Smarty) to render texts from templates stored in DB. For example, store email message template in DB table and generate email body from it on the fly.

**Comment by klimov-paul** [comment]
Use old good `eval()`.

**Comment by nineinchnick** [comment]
@klimov-paul why would you ever recommend `eval()`? @RomeroMsk you're asking what to use instead of `render()` in a controllers action?

**Comment by RomeroMsk** [comment]
@klimov-paul, are you serious?
@nineinchnick, yes I need to user `render()` method (or something like `renderString()`) with string content instead of view file path.

**Comment by nineinchnick** [comment]
I've updated my response above, use [renderContent](http://www.yiiframework.com/doc-2.0/yii-base-controller.html#renderContent%28%29-detail).

**Comment by klimov-paul** [comment]
> I've updated my response above, use renderContent.

I can not see how `renderContent()` can help here. It will just pass the PHP code as a plain string into the layout file.

As far as I understand you with to evaludate some string, stored in DB, as a view file content. There is no way for it instead of using `eval()`.
As far as I remember Smarty has built in `eval` function (or plugin), which allows evalution of a string as Smarty template content. As for Twig - I am unsure. For plain PHP view - native `eval()` is only option.

**Comment by RomeroMsk** [comment]
@nineinchnick, can you show an example of how to use this with Twig and not in controller context? Let's say we have `$template = 'Hello, {{ username }}!'` and want to put render result into `$result`, using `['username' => 'Test']` as parameters.

**Comment by RomeroMsk** [comment]
@klimov-paul, I've created this issue to ask for an ability to use stirng content instead of files in ViewRenderer. I don't like your workaround.

**Comment by klimov-paul** [comment]
This is not walk around - this is the only solution. Of course we can instoduce another method in the `View` class, but it will do just the same thing.
For smarty you can use following:
http://www.smarty.net/docs/en/language.function.eval.tpl
For Twig - I am don't know.

I suppose for your case a special tool should be used like following:
https://mustache.github.io/

**Comment by nineinchnick** [comment]
@RomeroMsk I think this does not concern Yii at all, you're asking how to process a Twig template. See [twig docs](http://twig.sensiolabs.org/doc/api.html).

**Comment by RomeroMsk** [comment]
@klimov-paul, yes, Mustache is good for this, but it's logic-less. Thanks, anyway.
@nineinchnick, I think you're right, thank you.

**Comment by lynicidn** [comment]
https://github.com/yiisoft/yii2/issues/6252

**Comment by baijunzhen** [comment]
@RomeroMsk 你可以这样
Yii::t('yii','you email body {username},send at{date}',['username'=>'foo',‘date’=>date('Y-m-d')]);
